### ISSUE TITLE
Build a proper connection between API metadata and organization

### ISSUE THREAD

**Issue by bajiat** [Digipalvelutehdas/APIKA#163]

**Comment by jykae** [comment]
Interested

**Comment by jykae** [comment]
```Does the organization data belong in metadata?```
Opinion: On higher level terms, yes, but not on database level, as we are currently creating more flexible database model for organizations feature.
Result: Selecting "if not" path, providing design options to choose how to show organization in API profile.

**Comment by jykae** [comment]
In the long run good modelling on database level could be Organizations, Contact/UserProfiles, Services. Organizations can have 1 or more Services, Services can have 1 or more APIs. UserProfile would be connected optionally(?) to Meteor User.

@brylie architectural view about above ? As for now we made binding Organization - Apis, I would make only separation of Organization from Metadata as part of this task.

**Comment by jykae** [comment]
![metadata 1](https://cloud.githubusercontent.com/assets/293452/21516274/59faf382-ccdf-11e6-88ed-f0efe8a751dc.png)

**Comment by jykae** [comment]
Services above haven't been requested, and can be still nested to abstract Metadata collection imho, until need to manage Services comes to APIKA (if ever).

UserProfile would be better to be separate collection/schema than current nested, also recommended by MDG https://guide.meteor.com/accounts.html#dont-use-profile

UserProfile improvement would be also beneficial for Apinf. 
Service support would remain to be decided if it's needed in future APIKA.

**Comment by brylie** [comment]
Great work @jykae, the diagram helps to clarify the components.

I agree that we should enhance the user profile, so lets open a separate issue (in Apinf) to track the progress.

What are your thoughts on the diagram from the original Metadata feature request ([apinf #472](https://github.com/apinf/platform/issues/472))?

![API Metadata schema form X-Road](https://cloud.githubusercontent.com/assets/1927158/11168086/e1ca8e9c-8b88-11e5-92cc-9e0e2c1352f0.png)

**Comment by brylie** [comment]
>UserProfile would be better to be separate collection/schema than current nested, also recommended by MDG https://guide.meteor.com/accounts.html#dont-use-profile
>
> UserProfile improvement would be also beneficial for Apinf.

I agree that we should enhance the user profile. Lets open a separate issue (in Apinf) to track the progress.
